util/reggen/validate.py

```diff
@@ -1108,27 +1113,28 @@ def validate_multi(mreg, offset, addrsep, width, top):
         mreg["name"] = "MREG_" + hex(offset)
     else:
         mrname = mreg['name']
+
+    # multireg defaults
+    # Should probably create this structure somewhere else as part checking,
+    # then directly create the default value
+    defaults = {'regwen_multi': False, 'compact': True}
+    for entry in defaults:
+        if entry not in mreg:
+            mreg[entry] = defaults[entry]
```

Comment:
Rupert taught me useful function here.
`mreg.setdefault(entry, defaults[entry])` :)

Reply:
that's pretty cool.. didn't realize this existed. 
Do you think it makes sense we just add this default function as a part of validation?
For the required keys, they have to have a value anyways.  For the optional ones, we can just introduce a default.  That way it can all be handled in one place.